pkg/validate/business_case.go

```diff
@@ -0,0 +1,55 @@
+package validate
+
+import (
+	"errors"
+
+	"github.com/cmsgov/easi-app/pkg/apperrors"
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// checkUniqLifecycleCosts checks if there's any duplicates
+func checkUniqLifecycleCosts(costs models.EstimatedLifecycleCosts) (string, string) {
+	costMap := map[string]bool{}
+
+	for _, cost := range costs {
+		attribute := string(cost.Solution) + string(cost.Year) + string(cost.Phase)
+		if costMap[attribute] {
+			return "LifecycleCostPhase", "cannot have multiple costs for the same phase, solution, and year"
+		}
+		costMap[attribute] = true
+	}
+	return "", ""
+}
+
+// check the system intake status is submitted
+func checkSystemIntakeSubmitted(intake *models.SystemIntake) (string, string) {
+	if intake.Status != models.SystemIntakeStatusSUBMITTED {
+		return "SystemIntake", "must have already been submitted"
+	}
+	return "", ""
+}
+
+// BusinessCaseForCreation checks if it's a valid business case to create
+func BusinessCaseForCreation(businessCase *models.BusinessCase, intake *models.SystemIntake) error {
```

Comment:
I think you can avoid passing in the whole intake here and just pass the status you need.

Reply:
same with the authorization stuff around, probably eventually needing more information from the whole model.